homeassistant/components/utility_meter/services.yaml

```diff
@@ -1,35 +1,41 @@
 # Describes the format for available switch services
 
 reset:
+  name: Reset
   description: Resets the counter of an utility meter.
-  fields:
-    entity_id:
-      description: Name(s) of the utility meter to reset
-      example: "utility_meter.energy"
+  target:
 
 next_tariff:
+  name: Next Tariff
   description: Changes the tariff to the next one.
-  fields:
-    entity_id:
-      description: Name(s) of entities to reset
-      example: "utility_meter.energy"
+  target:
 
 select_tariff:
-  description: selects the current tariff of an utility meter.
+  name: Select Tariff
+  description: Selects the current tariff of an utility meter.
+  target:
   fields:
-    entity_id:
-      description: Name of the entity to set the tariff for
-      example: "utility_meter.energy"
     tariff:
+      name: Tariff
       description: Name of the tariff to switch to
       example: "offpeak"
+      required: true
+      selector:
+        text:
 
 calibrate:
-  description: calibrates an utility meter.
+  name: Calibrate
+  description: Calibrates a utility meter sensor.
+  target:
+    entity:
+      domain: sensor
+      integration: utility_meter
```

Comment:
This probably doesn't work, this selector now limits it to targets that are entities, of the `sensor` domain, that are created/provided by the `utility_meter` integration. 

But this integration doesn't provide sensor entities, right? It provides `utility_meter` entities...

Reply:
it provides both, the utility_meter creates sensor entities to track each of the defined tariffs.

The utility_meter state tracks which tariff is running currently and the sensors track how much energy/water/gas in that given tariff